core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -88,6 +89,7 @@ public class RESTSessionCatalog extends BaseSessionCatalog
     implements Configurable<Configuration>, Closeable {
   private static final Logger LOG = LoggerFactory.getLogger(RESTSessionCatalog.class);
   private static final String REST_METRICS_REPORTING_ENABLED = "rest-metrics-reporting-enabled";
+  private static final String REST_SNAPSHOT_LOADING_MODE = "rest-snapshot-loading-mode";
```

Comment:
I think this should just be `snapshot-loading-mode`.

The reason why we added the `rest-` prefix for metrics reporting is that we report metrics in multiple ways, one of which is to always report through the REST interface. Because you can set the metrics reporter and choose whether to send REST metrics separately, we needed to distinguish the option. But that's not a problem for snapshot loading.